src/caffeine/coarray_queries_s.f90

```diff
@@ -22,15 +23,32 @@
   end procedure
 
   module procedure prif_image_index
-    integer :: dim
-    integer(c_int) :: prior_size
-
-    image_index = 1 + sub(1) - coarray_handle%info%lcobounds(1)
-    prior_size = 1
-    do dim = 2, size(sub)
-      prior_size = prior_size * (coarray_handle%info%ucobounds(dim-1) - coarray_handle%info%lcobounds(dim-1) + 1)
-      image_index = image_index + (sub(dim) - coarray_handle%info%lcobounds(dim)) * prior_size
-    end do
+    integer :: dim, i
+    integer(c_int) :: prior_size, num_img
+    logical :: invalid_cosubscripts
+
+    invalid_cosubscripts = .false.
+
+    check_subscripts: do i = 1, size(sub)
+       if (sub(i) .lt. coarray_handle%info%lcobounds(i) .or. sub(i) .gt. coarray_handle%info%ucobounds(i)) then
+          invalid_cosubscripts = .true.
+          exit check_subscripts
+       end if
+    end do check_subscripts
+
+    if (.not. invalid_cosubscripts) then
+      image_index = 1 + sub(1) - coarray_handle%info%lcobounds(1)
+      prior_size = 1
+      do dim = 2, size(sub)
+        prior_size = prior_size * (coarray_handle%info%ucobounds(dim-1) - coarray_handle%info%lcobounds(dim-1) + 1)
+        image_index = image_index + (sub(dim) - coarray_handle%info%lcobounds(dim)) * prior_size
```

Comment:
The above calculation looks wrong to me.

Have you tested anything with corank > 1?